lib/src/json_schema/json_schema.dart

```diff
@@ -58,170 +64,541 @@ class JsonSchema {
+      .forEach((k, v) => _makeSchema('$_path/definitions/$k', v, (rhs) => _definitions[k] = rhs));
 
-  _getProperties(dynamic value) {
-    if (value is Map) {
-      value.forEach((property, subSchema) =>
-          _makeSchema('$_path/properties/$property', subSchema, (rhs) => _properties[property] = rhs));
-    } else {
-      throw _objectError('properties', value);
+  /// Validate, calulcate and set the value of the 'description' JSON Schema prop.
+  _setDescription(dynamic value) => _description = TypeValidators.string('description', value);
+
+  /// Validate, calulcate and set the value of the 'enum' JSON Schema prop.
+  _setEnum(dynamic value) => _enumValues = TypeValidators.uniqueList('enum', value);
+
+  /// Validate, calulcate and set the value of the 'exclusiveMaximum' JSON Schema prop.
+  _setExclusiveMaximum(dynamic value) => _exclusiveMaximum = TypeValidators.boolean('exclusiveMaximum', value);
+
+  /// Validate, calulcate and set the value of the 'exclusiveMinimum' JSON Schema prop.
+  _setExclusiveMinimum(dynamic value) => _exclusiveMinimum = TypeValidators.boolean('exclusiveMinimum', value);
+
+  /// Validate, calulcate and set the value of the 'format' JSON Schema prop.
+  _setFormat(dynamic value) => _format = TypeValidators.string('format', value);
+
+  /// Validate, calulcate and set the value of the 'id' JSON Schema prop.
+  _setId(dynamic value) => _id = TypeValidators.uri('id', value);
+
+  /// Validate, calulcate and set the value of the 'minimum' JSON Schema prop.
+  _setMinimum(dynamic value) => _minimum = TypeValidators.number('minimum', value);
+
+  /// Validate, calulcate and set the value of the 'maximum' JSON Schema prop.
+  _setMaximum(dynamic value) => _maximum = TypeValidators.number('maximum', value);
+
+  /// Validate, calulcate and set the value of the 'maxLength' JSON Schema prop.
+  _setMaxLength(dynamic value) => _maxLength = TypeValidators.nonNegativeInt('maxLength', value);
+
+  /// Validate, calulcate and set the value of the 'minLength' JSON Schema prop.
+  _setMinLength(dynamic value) => _minLength = TypeValidators.nonNegativeInt('minLength', value);
+
+  /// Validate, calulcate and set the value of the 'multiple' JSON Schema prop.
+  _setMultipleOf(dynamic value) => _multipleOf = TypeValidators.nonNegativeNum('multiple', value);
+
+  /// Validate, calulcate and set the value of the 'not' JSON Schema prop.
+  _setNot(dynamic value) => _makeSchema('$_path/not', TypeValidators.object('not', value), (rhs) => _notSchema = rhs);
+
+  /// Validate, calulcate and set the value of the 'oneOf' JSON Schema prop.
+  _setOneOf(dynamic value) => _validateListOfSchema('oneOf', value, (schema) => _oneOf.add(schema));
+
+  /// Validate, calulcate and set the value of the 'pattern' JSON Schema prop.
+  _setPattern(dynamic value) => _pattern = new RegExp(TypeValidators.string('pattern', value));
+
+  /// Validate, calulcate and set the value of the 'ref' JSON Schema prop.
```

Comment:
#nit `calulcate` typo on all of these comments :)

Reply:
🤣